rosbag2_converter_default_plugins/src/rosbag2_converter_default_plugins/cdr/cdr_converter.cpp

```
@@ -63,37 +62,38 @@ std::string get_package_library_path(const std::string & package_name)
 CdrConverter::CdrConverter()
 {
   auto library_path = get_package_library_path("rmw_fastrtps_cpp");
-  std::shared_ptr<Poco::SharedLibrary> library;
   try {
-    library = std::make_shared<Poco::SharedLibrary>(library_path);
-  } catch (Poco::LibraryLoadException &) {
+    library = std::make_shared<rcpputils::SharedLibrary>(library_path);
+  } catch (std::runtime_error &) {
     throw std::runtime_error(
             std::string("poco exception: library could not be found:") + library_path);
+  } catch (std::bad_alloc &) {
```

Why catch and just let it throw?